src/client.ts

```diff
@@ -0,0 +1,132 @@
+import * as request from "request-promise-native";
+
+interface assoc {
+    [key: string]: any;
+}
+
+export class Client {
+
+    endpoint: string = 'https://appwrite.io/v1';
+    headers: assoc = {
+        'content-type': '',
+        'x-sdk-version': 'appwrite:typescript:',
+    };
+    
+    /**
+     * Set Project
+     *
+     * Your project ID
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setProject(value: string): this {
+        this.addHeader('X-Appwrite-Project', value);
+
+        return this;
+    }
+
+    /**
+     * Set Key
+     *
+     * Your secret API key
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setKey(value: string): this {
+        this.addHeader('X-Appwrite-Key', value);
+
+        return this;
+    }
+
+    /**
+     * Set Locale
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setLocale(value: string): this {
+        this.addHeader('X-Appwrite-Locale', value);
+
+        return this;
+    }
+
+    /**
+     * Set Mode
+     *
+     * @param string value
+     *
+     * @return self
+     */
+    setMode(value: string): this {
+        this.addHeader('X-Appwrite-Mode', value);
+
+        return this;
+    }
+
+
+    /***
+     * @param endpoint
+     * @return this
+     */
+    setEndpoint(endpoint: string): this {
+        this.endpoint = endpoint;
+
+        return this;
+    }
+
+    /**
+     * @param key string
+     * @param value string
+     */
+    addHeader(key: string, value: string): this {
+        this.headers[key.toLowerCase()] = value.toLowerCase();
```

Comment:
wouldn't lowercasing values introduce issues if it'd lowercase an intentionally-non-lowercase value?